collector/src/main/java/com/bitmovin/analytics/BitmovinAnalytics.java

```diff
@@ -229,6 +229,28 @@ public void onVideoChange() {
         Log.d(TAG, "onVideoChange");
     }
 
+  @Override
+  public void onSubtitleChange() {
+    Log.d(TAG, String.format("onAudioTrackChange %s", playerStateMachine.getImpressionId()));
+    EventData data = playerAdapter.createEventData();
+    data.setState(playerStateMachine.getCurrentState().toString().toLowerCase());
+    data.setDuration(0);
```

Comment:
AFAIK this will cause the state machine to call the `onExitState` from the `PLAYING` state - so it will send a playing-sample with the duration up to now (`onExitState` passes the duration as `timeStamp - machine.getOnEnterStateTimeStamp()`). Then it will call the `onEnterState` from the `SUBTITLECHANGE`, which will update the `enterStateTimeStamp` for succeeding state changes.
Next thing, going back to the `PLAYING` state again will cause the `onExitState` from `SUBTITLECHANGE` to be called, which will send a 0-duration sample indicating that a subtitle change occurred, and immediately enter the old state `PLAYING` again - `enterStateTimeStamp` will be updated again and the next duration will be fine again. So all good.
The only problem we're having (same as in Web): Sending a sample will take the current language from the `BitmovinAdapter` - which in this case will already be the new one. So the old `playing` sample, which had the old language, will be send with the new one. This will have to be addressed for everything else as well.

Reply:
So the state transitions are actually correct and sending the "old" sample with the new language is actually a known issue :/

Reply:
We should however ix this issue (maybe thoroughly thinking through state transitions once again :)